tests/integration/test_katib_experiments.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Integration tests for Katib Experiments.
+"""
+
+import logging
+
+import lightkube
+import pytest
+from pytest_operator.plugin import OpsTest
+from utils import (
+    assert_deleted,
+    assert_exp_status_running_succeeded,
+    assert_get_experiment,
+    create_experiment,
+    delete_experiment,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.parametrize(
+    "experiment_file",
+    [
+        "tests/assets/crs/bayesian-optimization.yaml",
+        "tests/assets/crs/cmaes.yaml",
+        "tests/assets/crs/darts-cpu.yaml",
+        "tests/assets/crs/enas-cpu.yaml",
+        "tests/assets/crs/file-metrics-collector.yaml",
+        "tests/assets/crs/grid-example.yaml",
+        "tests/assets/crs/hyperband.yaml",
+        "tests/assets/crs/median-stop.yaml",
+        "tests/assets/crs/random.yaml",
+        "tests/assets/crs/simple-pbt.yaml",
+    ],
+)
+async def test_kaitb_experiments(ops_test: OpsTest, experiment_file):
+    """Test Katib Experiments.
+
+    Create experiment and assert that it is Running or Succeeded.
+    At the end, experiment is deleted.
+    NOTE: This test is re-using deployment created in test_deploy_katib_charms() in test_charms.py
+    """
+    namespace = ops_test.model_name
+    lightkube_client = lightkube.Client()
+
+    # Add metrics collector injection enabled label to namespace
+    test_namespace = lightkube_client.get(
+        res=lightkube.resources.core_v1.Namespace, name=namespace
+    )
+    test_namespace.metadata.labels.update(
+        {"katib.kubeflow.org/metrics-collector-injection": "enabled"}
+    )
```

Comment:
Who would set this label normally? Why do we have to set it here? Isn't it part of our deployments?

Reply:
the profiles operator would usually set it, [see here](https://github.com/canonical/kubeflow-profiles-operator/blob/track/1.7/src/templates/namespace-labels.yaml#L16). I don't think we want to include profiles in these tests. WDYT?

Reply:
This is tied to the discussion about creating a namespace to run these tests in. I don't think we should ever create namespaces directly, since in Kubeflow they are meant to be managed by Profiles. And yes, it's the Profile Controller that sets the configured labels in the created namespaces. Is there a reason why we would not want to use Profiles in the tests?

Reply:
I understand we haven't tackled this yet, have we?

Reply:
I created a tech topic to discuss how we do it in other charms, and for this one I'll push changes that deploy profiles as you suggested.

Reply:
it's the safer choice until we come to an agreement.